algorithms/linfa-clustering/src/gaussian_mixture/algorithm.rs

```diff
@@ -293,7 +293,7 @@ impl<F: Float + Lapack + Scalar> GaussianMixtureModel<F> {
         let n_samples = observations.nrows();
         let (weights, means, covariances) = Self::estimate_gaussian_parameters(
             &observations,
-            &log_resp.mapv(|v| v.exp()),
+            &log_resp.mapv(|v| Scalar::exp(v)),
```

Comment:
can you import the trait and use the original order? There are several places where this becomes really long

Reply:
I am not sure to understand... What do you mean by the original order? Here if I use `v.exp()` I get the following compilation error:
```bash
   Compiling linfa-clustering v0.3.1 (D:\rlafage\workspace\linfa\algorithms\linfa-clustering)
error[E0034]: multiple applicable items in scope
    --> algorithms\linfa-clustering\src\gaussian_mixture\algorithm.rs:296:34
     |
296  |             &log_resp.mapv(|v| v.exp()),
     |                                  ^^^ multiple `exp` found
     |
note: candidate #1 is defined in the trait `ndarray_rand::rand_distr::num_traits::Float`
    --> D:\rlafage\.cargo\registry\src\github.com-1ecc6299db9ec823\num-traits-0.2.14\src\float.rs:1336:5
     |
1336 |     fn exp(self) -> Self;
     |     ^^^^^^^^^^^^^^^^^^^^^
note: candidate #2 is defined in the trait `Scalar`
    --> D:\rlafage\.cargo\registry\src\github.com-1ecc6299db9ec823\cauchy-0.3.0\src\lib.rs:112:5
     |
112  |     fn exp(self) -> Self;
     |     ^^^^^^^^^^^^^^^^^^^^^
help: disambiguate the associated function for candidate #1
     |
296  |             &log_resp.mapv(|v| ndarray_rand::rand_distr::num_traits::Float::exp(v)),
     |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: disambiguate the associated function for candidate #2
     |
296  |             &log_resp.mapv(|v| Scalar::exp(v)),
     |                                ^^^^^^^^^^^^^^
```

Reply:
ah I see, these are conflicting because the first exponential is real only and the second is complex. I was confused by the name `Scalar` representing the complex as well as real numbers. Any idea why this ambiguity appears now?

Reply:
After a bit of digging, I found that `ndarray-linalg` undergoes the same phenomenon : https://github.com/rust-ndarray/ndarray-linalg/commit/5157b22b0d3c911da7ad25f93037cd24dcc476f1 which led me to compare cauchy 0.2 vs 0.3 https://github.com/rust-math/cauchy/compare/0.2.2...0.3.0 and it seems that operations like `exp()` were changed to take `self` (instead of `&self`) which could explain the ambiguity is popping up now.